### PR TITLE
[MRG+1] Clarify documentation of pyplot.draw

### PR DESCRIPTION
**Pull request by Rufflewind**
See #7432.

### CODE REVIEW

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -674,10 +674,11 @@ def clf():
 def draw():
     """Redraw the current figure.
 
-    This is used in interactive mode to update a figure that has been
-    altered, but not automatically re-drawn.  This should be only rarely
-    needed, but there may be ways to modify the state of a figure with
-    out marking it as `stale`.  Please report these cases as bugs.
+    This is used to update a figure that has been altered, but not
+    automatically re-drawn.  If interactive mode is on (:func:`.ion()`), this
```
Comment (QuLogic) [inline lib/matplotlib/pyplot.py]: The parentheses are not really necessary, but it looks like that doesn't break the link.

**Comment by tacaswell** [comment]
The appveyor failure looks like it is conda-version specific.

**Comment by tacaswell** [comment]
backported to v2.x as 5ee1eee

**Comment by tacaswell** [comment]
@Rufflewind Congratulations on you first mpl contribution :tada: 

It is hard to stress how important documentation changes like this are, hopefully we will hear from you again!

**Comment by Rufflewind** [comment]
Thanks for merging and handling the issue quickly :)

**Pull request merged**